src/hints/execution.rs

```diff
@@ -1681,6 +1681,38 @@ pub fn get_old_block_number_and_hash(
     Ok(())
 }
 
+pub const COMPUTE_IDS_HIGH_LOW: &str = indoc! {r#"
+    from starkware.cairo.common.math_utils import as_int
+
+    # Correctness check.
+    value = as_int(ids.value, PRIME) % PRIME
+    assert value < ids.UPPER_BOUND, f'{value} is outside of the range [0, 2**165).'
+
+    # Calculation for the assertion.
+    ids.high, ids.low = divmod(ids.value, ids.SHIFT)"#
+};
+
+pub fn compute_ids_high_low(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let upper_bound = get_constant_from_var_name("UPPER_BOUND", constants)?;
+    let shift = get_constant_from_var_name("SHIFT", constants)?;
```

Comment:
I'm still not sure about using `get_constant_from_var_name` here. I think constants should appear in the `constants` HashMap passed here. Maybe what you want is vm/src/hint_processor/builtin_hint_processor/math_utils.rs, but either way look at the warning above that fn